container-engine-lib/lib/backend_impls/docker/docker_manager/docker_auth.go

```diff
@@ -0,0 +1,189 @@
+package docker_manager
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/docker/docker/api/types/registry"
+	dockerregistry "github.com/docker/docker/registry"
+	"github.com/kurtosis-tech/stacktrace"
+)
+
+const (
+	ENV_DOCKER_CONFIG string = "DOCKER_CONFIG"
+)
+
+// RegistryAuthConfig holds authentication configuration for a container registry
+type RegistryAuthConfig struct {
+	Auths       map[string]registry.AuthConfig `json:"auths"`
+	CredHelpers map[string]string              `json:"credHelpers"`
+	CredsStore  string                         `json:"credsStore"`
+}
+
+// loadDockerAuth loads the authentication configuration from the config.json file located in $DOCKER_CONFIG or ~/.docker
+func loadDockerAuth() (RegistryAuthConfig, error) {
+	configFilePath := os.Getenv(ENV_DOCKER_CONFIG)
+	if configFilePath == "" {
+		configFilePath = os.Getenv("HOME") + "/.docker/config.json"
+	} else {
+		configFilePath = configFilePath + "/config.json"
+	}
+
+	file, err := os.ReadFile(configFilePath)
+	if errors.Is(err, os.ErrNotExist) {
+		// If the auth config doesn't exist, return an empty auth config
```

Comment:
@skylenet I don't think Kurtosis should require the config json, so if it doesn't exist well just return empty auth config. What do you think? Otherwise engine won't start unless it exists.

Reply:
Yup! Nice catch